pkg/virt-controller/watch/vm.go

```diff
@@ -28,138 +28,149 @@ func NewVMControllerWithListWatch(vmService services.VMService, _ record.EventRe
+	// Fetch the latest Vm state from cache
+	obj, exists, err := store.GetByKey(key.(string))
 
-		// Fetch the latest Vm state from cache
-		obj, exists, err := store.GetByKey(key.(string))
+	if err != nil {
+		queue.AddRateLimited(key)
+		return
+	}
 
+	// Retrieve the VM
+	var vm *v1.VM
+	if !exists {
+		_, name, err := cache.SplitMetaNamespaceKey(key.(string))
 		if err != nil {
+			// TODO do something more smart here
 			queue.AddRateLimited(key)
 			return
 		}
+		vm = v1.NewVMReferenceFromName(name)
+	} else {
+		vm = obj.(*v1.VM)
+	}
+	logger := logging.DefaultLogger().Object(vm)
 
-		// Retrieve the VM
-		var vm *v1.VM
-		if !exists {
-			_, name, err := cache.SplitMetaNamespaceKey(key.(string))
-			if err != nil {
-				// TODO do something more smart here
-				queue.AddRateLimited(key)
-				return
-			}
-			vm = v1.NewVMReferenceFromName(name)
-		} else {
-			vm = obj.(*v1.VM)
+	if !exists {
+		// Delete VM Pods
+		err := vmd.vmService.DeleteVMPod(vm)
+		if err != nil {
+			logger.Error().Reason(err).Msg("Deleting VM target Pod failed.")
 		}
-		logger := logging.DefaultLogger().Object(vm)
-
-		if !exists {
-			// Delete VM Pods
-			err := vmService.DeleteVMPod(vm)
-			if err != nil {
-				logger.Error().Reason(err).Msg("Deleting VM target Pod failed.")
-			}
-			logger.Info().Msg("Deleting VM target Pod succeeded.")
-		} else if vm.Status.Phase == "" {
-			// Schedule the VM
-			vmCopy := v1.VM{}
-
-			// Deep copy the object, so that we can safely manipulate it
-			model.Copy(&vmCopy, vm)
-			logger := logging.DefaultLogger().Object(&vmCopy)
-
-			// Create a pod for the specified VM
-			//Three cases where this can fail:
-			// 1) VM pods exist from old definition // 2) VM pods exist from previous start attempt and updating the VM definition failed
-			//    below
-			// 3) Technical difficulties, we can't reach the apiserver
-			// For case (1) this loop is not responsible. virt-handler or another loop is
-			// responsible.
-			// For case (2) we want to delete the VM first and then start over again.
-
-			// TODO move defaulting to virt-api
-			if vmCopy.Spec.Domain == nil {
-				spec := v1.NewMinimalDomainSpec(vmCopy.GetObjectMeta().GetName())
-				vmCopy.Spec.Domain = spec
-			}
-			vmCopy.Spec.Domain.UUID = string(vmCopy.GetObjectMeta().GetUID())
-			vmCopy.Spec.Domain.Devices.Emulator = "/usr/local/bin/qemu-x86_64"
-			vmCopy.Spec.Domain.Name = vmCopy.GetObjectMeta().GetName()
-
-			// TODO when we move this to virt-api, we have to block that they are set on POST or changed on PUT
-			graphics := vm.Spec.Domain.Devices.Graphics
-			for i, _ := range graphics {
-				if strings.ToLower(graphics[i].Type) == "spice" {
-					graphics[i].Port = int32(4000) + int32(i)
-					graphics[i].Listen = v1.Listen{
-						Address: "0.0.0.0",
-						Type:    "address",
-					}
-
+		logger.Info().Msg("Deleting VM target Pod succeeded.")
+	} else if vm.Status.Phase == "" {
```

Comment:
Instead of "", we should use the constant v1.MigrationUnknown from the MigrationPhase constants